PebbleSharp.Core/Enums.cs

```diff
@@ -84,11 +68,166 @@ public enum LogLevel
         Verbose = 250
     }
 
-    public enum AppMessage : byte
+    public enum AppMessageCommand : byte
     {
         Push = 0x01,
         Request = 0x02,
         Ack = 0xFF,
         Nack = 0x7F
     }
+
+	public enum BlobDatabase:byte
+	{
+		Test = 0,
+		Pin = 1,
+		App = 2,
+		Reminder = 3,
+		Notification = 4
+	}
+
+	public enum BlobStatus:byte
+	{
+		Success = 0x01,
+		GeneralFailure = 0x02,
+		InvalidOperation = 0x03,
+		InvalidDatabaseID = 0x04,
+		InvalidData = 0x05,
+		KeyDoesNotExist = 0x06,
+		DatabaseFull = 0x07,
+		DataStale = 0x08
+	}
+
+	public enum BlobCommand:byte
+	{
+		Insert=0x01,
+		Delete=0x04,
+		Clear=0x05,
+	}
+
+	public enum AppRunState : byte
+	{
+		Start=0x01,
+		Stop=0x02,
+		Request=0x03,
+	}
+
+	public enum AppFetchStatus : byte
+	{
+		Start = 0x01,
+		Busy = 0x02,
+		InvalidUUID = 0x03,
+		NoData = 0x04
+	}
+
+
+	public enum TransferType : byte
+	{
+		Firmware = 1,
+		Recovery = 2,
+		SysResources = 3,
+		Resources = 4,
+		Binary = 5,
+		File=6,
+		Worker=7
+	}
+
+	public enum SystemMessage : byte
+	{
+		FirmwareAvailible = 0,
+		FirmwareStart = 1,
+		FirmwareComplete = 2,
+		FirmwareFail = 3,
+		FirmwareUpToDate = 4,
+		FirmwareOutOfDate = 5,
+		BluetoothStartDiscoverable = 6,
+		BluetoothEndDiscoverable = 7
+	}
+
+	public enum PutBytesType : byte
+	{
+		Init=0x01,
+		Put=0x02,
+		Commit=0x03,
+		Abort=0x04,
+		Install=0x05,
+	}
+
+	public enum PutBytesResult : byte
+	{
+		Ack=0x01,
+		Nack=0x02
+	}
+
+	public enum ResetCommand
+	{
+		Reset = 0x00,
+		DumpCore = 0x01,
+		FactoryReset = 0x02,
+		PRF = 0x03
+	}
+
+	public enum Hardware:byte
+	{
+		UNKNOWN = 0,
+		TINTIN_EV1 = 1,
+		TINTIN_EV2 = 2,
+		TINTIN_EV2_3 = 3,
+		TINTIN_EV2_4 = 4,
+		TINTIN_V1_5 = 5,
+		BIANCA = 6,
+		SNOWY_EVT2 = 7,
+		SNOWY_DVT = 8,
+		SPALDING_EVT = 9,
+		BOBBY_SMILES = 10,
+		SPALDING = 11,
+		TINTIN_BB = 0xFF,
+		TINTIN_BB2 = 0xFE,
+		SNOWY_BB = 0xFD,
+		SNOWY_BB2 = 0xFC,
+		SPALDING_BB2 = 0xFB,
+	}
+
+	public enum SoftwarePlatform : byte
+	{
+		UNKNOWN,
+		APLITE,
+		BASALT,
+		CHALK
+	}
+
+	public static class HardwareHelpers
+	{
+		private static Dictionary<Hardware, SoftwarePlatform> Platforms;
+
+		private static void Initialize()
```

Comment:
I struggle with this one only because I find the exceptions thrown by static constructors to be maddening to troubleshoot when they throw exceptions, that was the only reason I wrote it the way I did.